mountpoint-s3/src/fs.rs

```diff
@@ -604,15 +612,31 @@ where
         atime: Option<OffsetDateTime>,
         mtime: Option<OffsetDateTime>,
         _flags: Option<u32>,
+        size: Option<u64>,
     ) -> Result<Attr, Error> {
         tracing::info!(
-            "fs:setattr with ino {:?} flags {:?} atime {:?} mtime {:?}",
+            "fs:setattr with ino {:?} flags {:?} atime {:?} mtime {:?} size {:?}",
             ino,
             _flags,
             atime,
-            mtime
+            mtime,
+            size
         );
-        let lookup = self.superblock.setattr(&self.client, ino, atime, mtime).await?;
+        let lookup = match (self.superblock.setattr(&self.client, ino, atime, mtime).await, size) {
```

Comment:
please move the `setattr` call above the match, and match on a variable like `setattr_result`